src/EnergyPlus/ExhaustAirSystemManager.cc

```diff
@@ -0,0 +1,966 @@
+        auto &ip = state.dataInputProcessing->inputProcessor;
+        auto const instances = ip->epJSON.find(cCurrentModuleObject);
+        if (instances != ip->epJSON.end()) {
+            auto const &objectSchemaProps = ip->getObjectSchemaProps(state, cCurrentModuleObject);
+            auto &instancesValue = instances.value();
+            int numExhaustSystems = instancesValue.size();
+            int exhSysNum = 0;
+
+            if (numExhaustSystems > 0) {
+                state.dataZoneEquip->ExhaustAirSystem.allocate(numExhaustSystems);
+            }
+
+            for (auto instance = instancesValue.begin(); instance != instancesValue.end(); ++instance) {
+                ++exhSysNum;
+                auto const &objectFields = instance.value();
+                auto &thisExhSys = state.dataZoneEquip->ExhaustAirSystem(exhSysNum);
+                thisExhSys.Name = UtilityRoutines::MakeUPPERCase(instance.key());
+                ip->markObjectAsUsed(cCurrentModuleObject, instance.key());
+
+                std::string zoneMixerName = ip->getAlphaFieldValue(objectFields, objectSchemaProps, "airloophvac_zonemixer_name");
+                int zoneMixerIndex = 0;
+                bool zoneMixerErrFound = false;
+                EnergyPlus::MixerComponent::GetZoneMixerIndex(state, zoneMixerName, zoneMixerIndex, zoneMixerErrFound, thisExhSys.Name);
+
+                if (!zoneMixerErrFound) {
+                    // With the correct MixerNum Initialize
+                    EnergyPlus::MixerComponent::InitAirMixer(state, zoneMixerIndex); // Initialize all Mixer related parameters
+
+                    // See if need to do the zone mixer's CheckEquipName() function
+                    ValidateComponent(state, "AirLoopHVAC:ZoneMixer", zoneMixerName, IsNotOK, "AirLoopHVAC:ExhaustSystem");
+                    if (IsNotOK) {
+                        ShowSevereError(state, RoutineName + cCurrentModuleObject + "=" + thisExhSys.Name);
+                        ShowContinueError(state, "ZoneMixer Name =" + zoneMixerName + " mismatch or not found.");
+                        ErrorsFound = true;
+                    } else {
+                        // normal conditions
+                    }
+                } else {
+                    ShowSevereError(state, RoutineName + cCurrentModuleObject + "=" + thisExhSys.Name);
+                    ShowContinueError(state, "ZoneMixer Name =" + zoneMixerName + "not found.");
+                    ErrorsFound = true;
+                }
+                thisExhSys.ZoneMixerName = zoneMixerName;
+                thisExhSys.ZoneMixerIndex = zoneMixerIndex;
+
+                std::string centralFanType = ip->getAlphaFieldValue(objectFields, objectSchemaProps, "fan_object_type");
+                int centralFanTypeNum = 0;
+                // 2022-01: Check fan types and gives warnings
+                if (UtilityRoutines::SameString(centralFanType, "Fan:SystemModel")) {
```

Comment:
Use `getEnumerationValue` here.

Reply:
The `getEnumerationValue` may not apply here, since the two fan type numbers seem to be "global" int const defined in name space DataHVACGlobals--did not find a string array to enumerate those. Or do I missed the string array somewhere?